proxy/http/HttpSM.cc

```diff
@@ -178,7 +178,7 @@ HttpVCTable::find_entry(VIO *vio)
 //      for reuse
 //
 void
-HttpVCTable::remove_entry(HttpVCTableEntry *e)
+HttpVCTable::remove_entry(HttpVCTableEntry *e, HttpSM *sm)
```

Comment:
If this is needed for various methods, perhaps `HttpVCTable` should be constructed with a reference to the `HttpSM`.